### PR TITLE
jflap: refactoring, add xdg desktop item

### PR DESCRIPTION
**Pull request by yuuyins**
###### Description of changes
jflap: refactoring, add xdg desktop item

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by grnnja — APPROVED** [review]
LGTM!

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 173905` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>jflap</li>
  </ul>
</details>

File: pkgs/applications/science/programming/jflap/default.nix [hunk 1]
```diff
@@ -1,32 +1,67 @@
...
+        "ComputerScience"
+        "DataVisualization"
+        "Engineering"
+        "Java"
+      ];
+    })
+  ];
 
   installPhase = ''
-    runHook preInstall
```
Comment (SuperSandro2000) [inline pkgs/applications/science/programming/jflap/default.nix:18]: We want to keep that and add postInstall hook.

File: pkgs/applications/science/programming/jflap/default.nix [hunk 2]
```diff
@@ -1,32 +1,67 @@
...
+      --add-flags "-jar $out/share/java/jflap.jar"
   '';
 
   meta = with lib; {
     description = "GUI tool for experimenting with formal languages topics";
     homepage = "https://www.jflap.org/";
     license = licenses.unfree;
-    maintainers = [ maintainers.grnnja ];
+    maintainers = with maintainers; [ grnnja yuu ];
     platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/programming/jflap/default.nix:65]: Thats wrong. It is at most jre9.meta.platforms.

**Comment by yuuyins** [comment]
i moved the directory for https://github.com/NixOS/nixpkgs/issues/164019

the programming and computer-architecture should be probably merge into there as well which is all about computer science/engineering / software engineering

**Comment by SuperSandro2000** [comment]
> the programming and computer-architecture should be probably merge into there as well which is all about computer science/engineering / software engineering

not super important

**Pull request merged**